examples/model_selection/plot_grid_search_refit_callable.py

```diff
@@ -88,40 +120,220 @@ def best_low_complexity(cv_results):
+    .then(pl.lit("Selected"))
+    .otherwise(pl.lit("Regular"))
+    .alias("model_type")
+)
+
+# Create a figure with two subplots: one for test scores and one for train vs
+# test comparison
+fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(15, 6))
+
+# Plot 1: Test scores with error bars
+components = results_df["n_components"]
+colors = ["green" if comp == best_components else "skyblue" for comp in components]
+
+bars = ax1.bar(
+    components,
+    results_df["mean_test_score"],
+    width=1.0,
+    color=colors,
+    yerr=results_df["std_test_score"],
+    capsize=10,
+)
+
+# Add threshold lines to the first subplot
+ax1.axhline(
+    np.max(test_scores), linestyle="--", color="orange", label="Best score", linewidth=2
+)
+ax1.axhline(lower, linestyle="--", color="red", label="Best score - 1 std", linewidth=2)
 
-print("The best_index_ is %d" % best_index_)
-print("The n_components selected is %d" % n_components[best_index_])
-print(
-    "The corresponding accuracy score is %.2f"
-    % grid.cv_results_["mean_test_score"][best_index_]
+# Plot 2: Train vs Test scores
+ax2.plot(
+    components,
+    results_df["mean_train_score"],
+    "o-",
+    color="green",
+    label="Train Score",
 )
+ax2.plot(
+    components,
+    results_df["mean_test_score"],
+    "o-",
+    color="blue",
+    label="Test Score",
+)
+
+# Add error bands for train and test scores
+for i, (comp, train_score, test_score, train_std, test_std) in enumerate(
+    zip(
+        components,
+        results_df["mean_train_score"],
+        results_df["mean_test_score"],
+        results_df["std_train_score"],
+        results_df["std_test_score"],
+    )
+):
+    ax2.fill_between(
+        [comp - 0.2, comp + 0.2],
+        [train_score - train_std, train_score - train_std],
+        [train_score + train_std, train_score + train_std],
+        color="green",
+        alpha=0.2,
+    )
+    ax2.fill_between(
+        [comp - 0.2, comp + 0.2],
+        [test_score - test_std, test_score - test_std],
+        [test_score + test_std, test_score + test_std],
+        color="blue",
+        alpha=0.2,
+    )
+
+# Mark the best model in the second plot
+ax2.plot(
+    best_components, best_score, "o", color="red", markersize=10, label="Selected model"
+)
+
+# Set titles and labels for both subplots
+ax1.set_title("Model Complexity vs. Test Score", fontsize=14)
+ax1.set_xlabel("Number of PCA components", fontsize=12)
+ax1.set_ylabel("Digit classification accuracy", fontsize=12)
+ax1.set_xticks(components)
+ax1.set_ylim((0.7, 1.0))  # Adjust as needed for better visualization
+ax1.grid(axis="y", linestyle="--", alpha=0.7)
+ax1.legend(loc="lower right")
+
+ax2.set_title("Train vs. Test Performance", fontsize=14)
+ax2.set_xlabel("Number of PCA components", fontsize=12)
+ax2.set_ylabel("Score", fontsize=12)
+ax2.set_xticks(components)
+ax2.set_ylim((0.7, 1.0))  # Adjust as needed for better visualization
+ax2.grid(axis="both", linestyle="--", alpha=0.7)
+ax2.legend(loc="lower right")
```

Comment:
I am also not really a plotting person, the only thing I can think that shortens it (but honestly not sure it is better) is:

```suggestion
for ax in (ax1, ax2):
    ax.set_xlabel("Number of PCA components", fontsize=12)
    ax.set_xticks(components)
    ax.set_ylim((0.7, 1.0))  # Adjust as needed for better visualization
    ax.legend(loc="lower right")
       
ax1.set_title("Model Complexity vs. Test Score", fontsize=14)
ax1.set_ylabel("Digit classification accuracy", fontsize=12)
ax1.grid(axis="y", linestyle="--", alpha=0.7)

ax2.set_title("Train vs. Test Performance", fontsize=14)
ax2.set_ylabel("Score", fontsize=12)
ax2.grid(axis="both", linestyle="--", alpha=0.7)
```